### PR TITLE
Add html site map in the sandbox

### PR DESCRIPTION
**Pull request by sampaccoud**
## Purpose

A first feedback from the accessibility audit we conducted is that we are missing a "second way to navigate the site". A best practice is to add a page with an HTML sitemap.

## Proposal

Add a new plugin to easily create sitemaps.

This is a 2-storey parent/child plugin. 

The parent plugin is just a container. The child plugin defines a subtree of the sitemap and allow a lot of configuration:

- make a sitemap for the whole site with a single instance,
- make a subtree including or not the targeted page,
- limit the depth of each subtree displayed in the sitemap,
- show or not the pages that are excluded from navigation,
- show pages requiring login only to the authenticated users,
- on a public page, show only the public pages that are published,
  in the current language.

### CODE REVIEW

File: src/richie/plugins/html_sitemap/cms_plugins.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        else:
+            nb_open = max(pages[i].level - pages[i - 1].level, 0)
+
+        is_empty = True
+        if i == nb_pages - 1:
+            nb_close = pages[i].level - 1
+        else:
+            gap = pages[i].level - pages[i + 1].level
+            nb_close = max(pages[i].level - pages[i + 1].level, 0)
+            is_empty = gap >= 0
```
Comment (jmaupetit) [inline src/richie/plugins/html_sitemap/cms_plugins.py]: `is_empty` is disturbing for a page, what about `has_children` or `is_leaf`?

File: src/richie/plugins/html_sitemap/cms_plugins.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+        yield page, nb_open, nb_close, is_empty
```
[the remark this answers is not in the document]
Comment (sampaccoud) [reply src/richie/plugins/html_sitemap/cms_plugins.py]: True. This is from a previous implementation ;-)

File: src/richie/plugins/html_sitemap/cms_plugins.py [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+            nb_open = 0
+        else:
+            nb_open = max(pages[i].level - pages[i - 1].level, 0)
+
+        is_empty = True
+        if i == nb_pages - 1:
+            nb_close = pages[i].level - 1
+        else:
+            gap = pages[i].level - pages[i + 1].level
+            nb_close = max(pages[i].level - pages[i + 1].level, 0)
```
[the remark this answers is not in the document]
Comment (sampaccoud) [reply src/richie/plugins/html_sitemap/cms_plugins.py]: Nice! In the end this method is very simple ;-)

File: src/richie/plugins/html_sitemap/cms_plugins.py (hunk 1 above)
Comment (sampaccoud) [reply src/richie/plugins/html_sitemap/cms_plugins.py]: let's go back to `is_leaf`.

**Pull request merged**